python/dgl/data/tree.py

```diff
@@ -1,112 +1,181 @@
 """Tree-structured data.
-
 Including:
     - Stanford Sentiment Treebank
 """
 from __future__ import absolute_import
 
-from collections import namedtuple, OrderedDict
+from collections import OrderedDict
 import networkx as nx
 
 import numpy as np
 import os
 
+from .dgl_dataset import DGLBuiltinDataset
 from .. import backend as F
+from .utils import _get_dgl_url, save_graphs, save_info, load_graphs, \
+    load_info, deprecate_property
 from ..convert import from_networkx
-from .utils import download, extract_archive, get_download_dir, _get_dgl_url
-from ..utils import retry_method_with_fix
 
-__all__ = ['SSTBatch', 'SST']
+__all__ = ['SST', 'SSTDataset']
 
-_urls = {
-    'sst' : 'dataset/sst.zip',
-}
 
-SSTBatch = namedtuple('SSTBatch', ['graph', 'mask', 'wordid', 'label'])
+class SSTDataset(DGLBuiltinDataset):
+    r"""Stanford Sentiment Treebank dataset.
 
-class SST(object):
-    """Stanford Sentiment Treebank dataset.
+    .. deprecated:: 0.5.0
+        `trees` is deprecated, it is replaced by:
+            >>> dataset = SSTDataset()
+            >>> for tree in dataset:
+            ....    # your code here
+            ....
+            >>>
+        `num_vocabs` is deprecated, it is replaced by:
+            >>> dataset = SSTDataset()
```

Comment:
You can directly say: `num_vocabs` is deprecated, it is replaced by `vocab_size`